src/MultiRegion/multi_region_output_writers.jl

```diff
@@ -51,8 +52,11 @@ function fetch_output(csf::CubedSphereField, model)
     return parent(csf)
 end
 
-convert_output(mo::MultiRegionObject, writer) = 
-    MultiRegionObject(Tuple(convert(writer.array_type, obj) for obj in mo.regional_objects))
+convert_output(mo::MultiRegionObject, writer) =
+    MultiRegionObject(
+        architecture_from_type(writer.array_type),
+        Tuple(convert(writer.array_type, obj) for obj in mo.regional_objects)
+    )
```

Comment:
I think we should make this a multiline function to improve clarity, e.g.,

```suggestion
function convert_output(mo::MultiRegionObject, writer)
    array_type = writer.array_type
    arch = architecture_from_type(array_type)
    return MultiRegionObject(arch, Tuple(convert(array_type, obj) for obj in mo.regional_objects)
    )
end
```

although, the above won't work because of what's noted at https://github.com/CliMA/Oceananigans.jl/pull/4499#issuecomment-3046611434